tools/lsp/properties.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+use i_slint_compiler::diagnostics::Spanned;
+use i_slint_compiler::object_tree::{Element, ElementRc};
+
+#[derive(serde::Deserialize, serde::Serialize)]
+pub(crate) struct PropertyInformation {
+    name: String,
+    type_name: String,
+    declared_at: Option<(String, u32)>,
+    defined_at: Option<(u32, u32)>, // Range in the elements source file!
+}
+
+#[derive(serde::Deserialize, serde::Serialize)]
+pub(crate) struct QueryPropertyResponse {
+    properties: Vec<PropertyInformation>,
+    source_file: Option<String>,
+}
+
+// This gets defined accessibility properties...
+fn get_reserved_properties(result: &mut Vec<PropertyInformation>) {
+    result.extend(i_slint_compiler::typeregister::reserved_properties().map(|p| {
+        PropertyInformation {
+            name: p.0.to_string(),
+            type_name: format!("{}", p.1),
+            declared_at: None,
+            defined_at: None,
+        }
+    }))
+}
+
+fn source_file(element: &Element) -> Option<String> {
+    element.source_file().map(|sf| sf.path().to_string_lossy().to_string())
+}
+
+fn get_element_properties(element: &Element, result: &mut Vec<PropertyInformation>) {
```

Comment:
I'm curious, would not be possible to return a `impl Iterator` instead? I have the feeling that it might be more idiomatic and it has the added advantage that it would provide a size hint (at least for declarations).

Reply:
Yeap, I should do that.

Reply:
Done in next version.